IOSAccessAssessment/IOSAccessAssessmentApp.swift

```diff
@@ -9,20 +9,14 @@ import SwiftUI
 
 @main
 struct IOSAccessAssessmentApp: App {
-    private let keychainService = KeychainService()
-    @State private var isAuthenticated: Bool
-    
-    init() {
-        let isTokenValid = keychainService.isTokenValid()
-        _isAuthenticated = State(initialValue: isTokenValid)
-    }
+    @AppStorage("isAuthenticated") private var isAuthenticated: Bool = false
```

Comment:
I wonder if this way of setting up the authentication would be the best way to go. At the first glance, I do not see a reason to store authentication status in user defaults, when we can keep checking if the auth tokens are valid.  
May I be reminded on the incentive behind shifting to AppStorage instead of ViewModel?

Here are some doubts I have
1. Without a valid refresh token I am not sure if we can save any valid authentication. 
2. Even if we can enable user auth persistence, shouldn't we be continuously checking if the KeyChain still has valid credentials, instead of caching isAuthenticated to an insecure storage?